src/server/req_manager.c

```diff
@@ -650,6 +651,11 @@ set_queue_type(attribute *pattr, void *pque, int mode)
 				(((pbs_queue *)pque)->qu_attr[(int)QE_ATR_HasNodes].at_flags & ATR_VFLAG_SET) &&
 				(((pbs_queue *)pque)->qu_attr[(int)QE_ATR_HasNodes].at_val.at_long != 0)) {
 				return (PBSE_ATTRTYPE);
+			} else {
+				if (((pbs_queue *)pque)->qu_attr[(int)QA_ATR_partition].at_flags & ATR_VFLAG_SET &&
+						(spectype == QTYPE_RoutePush)) {
+					return PBSE_CANNOT_SET_ROUTE_QUE;
```

Comment:
The current function is action function only.  It was named as set_queue_type which may be causing the confusion. It could have been renamed to something like action_queue_type.